### ISSUE TITLE
Bad reasonPhrase when HttpStatus is provided for HttpMessage

### ISSUE THREAD

**Issue by mwojterski** [christophd/citrus#134]
In module `citrus-http` class `HttpMessage` has method `status(HttpStatus)` which reads `reasonPhrase` using `Enum#name()` instead `HttpStatus#getReasonPhrase()`:
```java
public HttpMessage status(HttpStatus statusCode) {
    // ...
    reasonPhrase(statusCode.name());
    // ...
}
```
As a consequence if one defines receive action with status defined via `ReceiveHttpMessageBuilder#status` (_where reason phrase is accessed correctly_) test will fail when it should pass.

**Comment by christophd** [comment]
This issue has already been fixed with dc91e7cc88f57e0ed01df715d435ebb60d7bb3ed in Citrus version 2.6 

We decided to do it the other way round not using the reason phrase but the enum name. You must be using a version older than 2.6

**Comment by mwojterski** [comment]
Yes, I'm using older version and haven't noticed that, sorry.

Btw, this seems like a strange decision, considering there is an exposed method `HttpClientResponseActionBuilder#reasonPhrase(String)`, which can be used to setup factual `reasonPhrase`. It can be confusing that it should in fact receive `HttpStatus` enum name and not what is commonly known as a `reasonPhrase`. But I guess you had your reasons (_no pun intended_).

I'm sorry for wasting your time, thanks for clarification and many thanks for citrus!
